packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/TransactionModal.tsx

```diff
@@ -0,0 +1,227 @@
+"use client";
+import * as DialogPrimitive from "@radix-ui/react-dialog";
+import { ArrowTopRightIcon } from "@radix-ui/react-icons";
+import cn from "classnames";
+import { PropsWithChildren, useContext, useState } from "react";
+
+import { Button } from "#/components";
+import { Toast } from "#/components/Toast";
+import { PoolMetadataContext } from "#/contexts/PoolMetadataContext";
+import { TransactionStatus, useTransaction } from "#/hooks/useTransaction";
+
+type Stage = {
+  pinColor: string;
+  confirmColor: string;
+  writeColor: string;
+};
+
+const STAGE_MAPPING: Record<TransactionStatus, Stage> = {
+  [TransactionStatus.PINNING]: {
+    pinColor: "border-yellow-400 text-yellow-400",
+    confirmColor: "bg-gray-400",
+    writeColor: "border-gray-400 text-gray-400",
+  },
+  [TransactionStatus.CONFIRMING]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-yellow-400 text-yellow-400",
+  },
+  [TransactionStatus.SUBMITTING]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-yellow-400 text-yellow-400",
+  },
+  [TransactionStatus.CONFIRMED]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-green-400 text-green-400",
+  },
+  [TransactionStatus.PINNING_ERROR]: {
+    pinColor: "border-red-400 text-red-400",
+    confirmColor: "bg-gray-400",
+    writeColor: "border-gray-400 text-gray-400",
+  },
+  [TransactionStatus.WRITE_ERROR]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-red-400 text-red-400",
+  },
+  [TransactionStatus.AUTHORIZING]: {
+    pinColor: "border-yellow-400 text-yellow-400",
+    confirmColor: "bg-gray-400",
+    writeColor: "border-gray-400 text-gray-400",
+  },
+  [TransactionStatus.WAITING_APPROVAL]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-yellow-400 text-yellow-400",
+  },
+};
+
+function ConfirmationAlert({
+  handleSubmit,
+}: {
+  handleSubmit: (value: boolean) => void;
```

Comment:
I don't think this definition is correct. You're not passing value to this function.